flag_ext.go

```diff
@@ -1,14 +1,74 @@
 package cli
 
-import "flag"
+import (
+	"context"
+	"flag"
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+var _ Value = (*externalValue)(nil)
+
+var (
+	_ Flag                        = (*extFlag)(nil)
+	_ ActionableFlag              = (*extFlag)(nil)
+	_ CategorizableFlag           = (*extFlag)(nil)
+	_ DocGenerationFlag           = (*extFlag)(nil)
+	_ DocGenerationMultiValueFlag = (*extFlag)(nil)
+	_ LocalFlag                   = (*extFlag)(nil)
+	_ RequiredFlag                = (*extFlag)(nil)
+	_ VisibleFlag                 = (*extFlag)(nil)
+)
+
+// -- Value Value
+type externalValue struct {
+	e *extFlag
+}
+
+// Below functions are to satisfy the flag.Value interface
+
+func (ev *externalValue) Set(s string) error {
+	if ev != nil && ev.e.f != nil {
+		return ev.e.f.Value.Set(s)
+	}
+	return nil
+}
+
+func (ev *externalValue) Get() any {
+	if ev != nil && ev.e.f != nil {
+		return ev.e.f.Value.(flag.Getter).Get()
+	}
+	return nil
+}
+
+func (ev *externalValue) String() string {
+	if ev != nil && ev.e.f != nil {
+		return ev.e.String()
+	}
+	return ""
+}
+
+func (ev *externalValue) IsBoolFlag() bool {
+	if ev == nil || ev.e.f == nil {
+		return false
+	}
+	bf, ok := ev.e.f.Value.(boolFlag)
+	return ok && bf.IsBoolFlag()
+}
 
 type extFlag struct {
-	f *flag.Flag
+	f        *flag.Flag
+	category string
 }
 
 func (e *extFlag) PreParse() error {
 	if e.f.DefValue != "" {
-		return e.Set("", e.f.DefValue)
+		// suppress errors for write-only external flags that always return nil
+		if err := e.Set("", e.f.DefValue); err != nil && e.f.Value.(flag.Getter).Get() != nil {
```

Comment:
```suggestion
		if err := e.Set("", e.f.DefValue); err != nil  {
```

do you need the Get() != nil check ?